### PR TITLE
The metamodel never uses more than 50% of the total time

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] The documentation is up-to-date with the changes I made.
- [ ] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [ ] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

**Comment by teytaud** [comment]
Can not work because not picklable.
We need something like that though.

**Comment by teytaud** [comment]
> Can not work because not picklable. We need something like that though.

Ok fixed.

**Review by jrapin — APPROVED** [review]
I don't quite like relying on multiprocessing but if you are confident with this...

File: nevergrad/optimization/optimizerlib.py [hunk 1]
```diff
@@ -1449,7 +1450,12 @@ class MetaModelFailure(ValueError):
     """Sometimes the optimum of the metamodel is at infinity."""
 
 
-def _learn_on_k_best(archive: utils.Archive[utils.MultiValue], k: int) -> tp.ArrayLike:
+def train_sklearn(model, X2, y, return_dict) -> None:
```
Comment (bottler) [inline nevergrad/optimization/optimizerlib.py:1453]:
```suggestion
def _train_sklearn(model, X2, y, return_dict) -> None:
```

**Comment by teytaud** [comment]
> I don't quite like relying on multiprocessing but if you are confident with this...

I would prefer another solution --- if there was one (?)...

File: nevergrad/optimization/optimizerlib.py (hunk 1 above)
Comment (teytaud) [reply nevergrad/optimization/optimizerlib.py]: Done.

**Comment by jrapin** [comment]
Have you tried the performance ? I'm a bit afraid of the impact it can have as well. Eg: how many time does the metamodel actually finishes? I guess it depends on the application

File: nevergrad/optimization/optimizerlib.py [hunk 2]
```diff
@@ -1449,7 +1451,12 @@ class MetaModelFailure(ValueError):
     """Sometimes the optimum of the metamodel is at infinity."""
 
 
-def _learn_on_k_best(archive: utils.Archive[utils.MultiValue], k: int) -> tp.ArrayLike:
+def _train_sklearn(model, X2, y, return_dict) -> None:
+    model.fit(X2, y)
+    return_dict["model"] = model
+
+
+def _learn_on_k_best(archive: utils.Archive[utils.MultiValue], k: int, timeout: int) -> tp.ArrayLike:
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:1459]: can we move both this functions to a different file? They are begginning to be big, and messy, in a file that is already big, and they should not serve as a model for later development...

Comment (jrapin) [reply nevergrad/optimization/optimizerlib.py:1459]: eg : optimization/utils.py

**Comment by teytaud** [comment]
Do i correctly remember that the use of multiprocesding here would be a problem and that therefore i should close ?

**Pull request closed without merging**